src/header/common/pragma.rs

```diff
@@ -0,0 +1,62 @@
+use std::fmt;
+use std::ascii::AsciiExt;
+
+use header::{Header, HeaderFormat, parsing};
+
+/// The `Pragma` header defined by HTTP/1.0.
+///
+/// > The "Pragma" header field allows backwards compatibility with
+/// > HTTP/1.0 caches, so that clients can specify a "no-cache" request
+/// > that they will understand (as Cache-Control was not defined until
+/// > HTTP/1.1).  When the Cache-Control header field is also present and
+/// > understood in a request, Pragma is ignored.
+
+/// > In HTTP/1.0, Pragma was defined as an extensible field for
+/// > implementation-specified directives for recipients.  This
+/// > specification deprecates such extensions to improve interoperability.
+///
+/// Spec: https://tools.ietf.org/html/rfc7234#section-5.4
+#[derive(Clone, PartialEq, Show)]
+pub enum Pragma{
+    /// Corresponds to the `no-cache` value.
+    NoCache,
+    /// Every value other than `no-cache`.
+    Ext(String),
+}
+
+impl Header for Pragma {
+    fn header_name(_: Option<Pragma>) -> &'static str {
+        "Pragma"
+    }
+
+    fn parse_header(raw: &[Vec<u8>]) -> Option<Pragma> {
+        parsing::from_one_raw_str(raw).and_then(|s: String| {
+            let c = s.to_ascii_lowercase();
+            let slice = &c.clone()[];
```

Comment:
Ok, bug in code. I want to preserve the upper/lowercase of the value, but the matching should be case insensitive. You will see it in the new code.